### PR TITLE
#M01. Judging automation - calculate and display total scores

### PR DESCRIPTION
**Pull request by natalijabujevic0708**
## Description
Updated the check_projects_scores function so If all the projects were scored, it calculates the total score for each team, sorts the teams by scores and renders final_score.html with the score table.
If all projects weren't scored, it renders final_score.html without the score table.

## Pull request type

#M01. Judging automation 

## Testing
I created Hackathon 1:

- teams: Team 1 with Project 1, Team 2 with Project Team 2 and Team 3 with Project Team 3
- judges: natalija, admin
- categories: Design, Function

**First scenario:**
1. Judge admin scored all the projects in all categories but judge natalija scored only teams 2 and 3

admin_2_score

2. After the judge natalija scored team 1, final_score.html is rendered with the score table

judging_2_score

table_2_scores

**==> if all the projects are scored,  it calculates the total score for each team, sorts the teams by scores and renders final_score.html with the score table**

**Testing the url path:**
If the url path is written with the specific hack_id it displays the same results
url3

**Second scenario:**
1. Judge admin scored all the projects in all categories but judge natalija scored only team 2

admin_4_scores

2. After the judge natalija scored team 1, final_score.html is rendered without the score table because the judge still needs to score team 3

judging_4_scores

table_4_scores

**==> If all projects weren't scored, it renders final_score.html without the score table**

**Testing the url path:**

If the url path is written with the specific hack_id it displays the same results

url

## Additional Information
There is one part of judging automation left to develop - to trigger the winning team's profile to allow them to update their shipping address.

## Does this introduce a breaking change

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by TravelTimN** [comment]
@natalijabujevic0708 
Is this also related to #112 presumably, yeah?

**Comment by natalijabujevic0708** [comment]
@TravelTimN yes, in #112 I just checked if all the judges scored all the projects, in this pull request I calculated the scores and displayed them!

File: hackathon/templates/hackathon/final_score.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{% extends "base.html" %}
+{% load static %}
+{% block content %}
+<h1 class="table_titles"> {{ hackathon }} <br> Final Scores </h1>
+	{% if sorted_teams_scores %}
+	<table class="final_score_table">
+		<tr class="table_titles">
```
Comment (TravelTimN) [inline hackathon/templates/hackathon/final_score.html:7]: Would it be more semantically acceptable to use `<thead>` and then `<tbody>` for this particular table?

File: hackathon/templates/hackathon/final_score.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+			<th> Project </th>
+			<th> Score </th>
+		</tr>
+		{% for team in sorted_teams_scores %}
+		<tr>
+			<td>{{ team.0}}</td>
```
Comment (TravelTimN) [inline hackathon/templates/hackathon/final_score.html:13]: Missing a minor "space" after 0

File: hackathon/templates/hackathon/final_score.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+			<td>{{ team.1 }}</td>
+		</tr>
+		{% endfor %}
+	</table>
+	{% else %}
+	<strong>There are no final scores for this Hackathon.</strong>
+	{% endif %}
+	{% endblock %}
```
Comment (TravelTimN) [inline hackathon/templates/hackathon/final_score.html:21]: File needs a blank line at the end.

File: static/css/style.css [hunk 4]
```diff
@@ -247,6 +247,35 @@ html .all-auth-container {
     text-align: left;
 
 }
+/* Final Score page */
+
+.table_titles + tr {
+    background-color: var(--p-orange);
+    color: white;
```
Comment (TravelTimN) [inline static/css/style.css:254]: Remember to use the CSS `:root` variable for `var(--white);` where applicable.

File: static/css/style.css [hunk 5]
```diff
@@ -247,6 +247,35 @@ html .all-auth-container {
     text-align: left;
 
 }
+/* Final Score page */
+
+.table_titles + tr {
```
Comment (TravelTimN) [inline static/css/style.css:252]: These targeted elements might need adjusted as well, if going for a more semantic `<thead>` and `<tbody>` format. (FYI)

**Review by TravelTimN — CHANGES_REQUESTED** [review]
@natalijabujevic0708 
Looking great! A few minor adjustments/comments, and should be grand!

**Comment by natalijabujevic0708** [comment]
@TravelTimN done :)

**Review by TravelTimN — APPROVED** [review]
LGTM

**Pull request merged**